### DISCUSSION TITLE
The issue of importing ACQ format files

### DISCUSSION THREAD

**Discussion by KD206** [bachlab/PsPM#801] [Q&A] [answered]
I am a beginner in PSPM and hope to use it to process and analyze SCR data. AndI am currently facing some difficulties regarding importing data. If possible, I hope to get your help.
At present, I can confirm that the format is acq version<3.9.0. My operation steps are shown in the following figure.
And it reports errors，which I show it in the picture.But I am not sure where the problem lies, whether it is a file name issue.
![屏幕截图 2024-11-21 174823](https://github.com/user-attachments/assets/0715979d-2867-482c-9dbc-a99947ac4e74)
![屏幕截图 2024-11-21 180834](https://github.com/user-attachments/assets/ca5680cf-6255-4cb7-ad52-511015d7731a)

**Comment by dominikbach** [comment]
Apparently `acqread` fails to read your file. To troubleshoot could you perhaps share the *.acq file (here or privately)?

**Reply by KD206** [reply]
I'm sorry, I just realized that my address doesn't seem to be displayed. I'm not sure if you can see the specific email address on your end. If not, perhaps I can share one of the examples here. Looking forward to your reply. And my personal address is 1257265383@qq.com Or 8304210402@csu.edu.cn.

**Reply by KD206** [reply]
Dear researcher, hello. On November 30th, I have already sent the file to the email address shown in the picture below. And I have not received a response yet, so I am not sure if you have received it or if the email address I sent is incorrect. Thank you for your help. If there is any disturbance, please forgive me. We look forward to your reply.
![屏幕截图 2024-12-04 210519](https://github.com/user-attachments/assets/38576812-0414-4244-bf0c-6e624377bc51)

**Reply by dominikbach** [reply]
Thank you, we haven't received anything yet. Please send to my personal email address (`www.caian.uni-bonn.de/en/team/team/dominik-r-bach`)

**Reply by KD206** [reply]
> Thank you, we haven't received anything yet. Please send to my personal email address (`www.caian.uni-bonn.de/en/team/team/dominik-r-bach`)

Dear developer, I have sent the specific files to your personal email. Thank you again for your help. If you haven't received the email, please let me know and I will send it to you again.

**Reply by dominikbach** [reply]
Thanks for sending the file. I suspect that this is not actually an Acknowledge file. I sent troubleshooting instructions by email.

**Comment by KD206** [comment]
Perhaps I should send it to you privately. So, can you share your email address. Or my personal address is ***@***.*** Or ***@***.***




君陌
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "bachlab/PsPM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月23日(星期六) 凌晨2:41
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [bachlab/PsPM] The issue of importing ACQ format files (Discussion #801)





 
Apparently acqread fails to read your file. To troubleshoot could you perhaps share the *.acq file (here or privately)?

**Comment by yooow-free** [comment]
Maybe I also have such a problem, can not import acq data. Here are my steps and error results.
![报错](https://github.com/user-attachments/assets/10eadd3e-30e7-4eb1-ab3a-09831611b8a9)
![步骤](https://github.com/user-attachments/assets/c8921176-b4e0-4af2-9238-a3936c5b3141)
![步骤1](https://github.com/user-attachments/assets/f89e5640-f857-4d85-928e-81199a117ca2)
![报错](https://github.com/user-attachments/assets/10eadd3e-30e7-4eb1-ab3a-09831611b8a9)

**Reply by dominikbach** [reply]
Could you also share a data file? You can find our email address here: https://www.caian.uni-bonn.de/en

**Reply by dominikbach** [reply]
Thanks for sending the file. I could not reproduce the problem, i.e. I was able to import the data file with SCR and digital marker channels. Unfortunately, I don't read Chinese and so cannot understand the warning messages that you posted.

**Comment by yooow-free** [comment]
Dear Prof. Dominik,
Thanks for your help. The warning about matlab is translated as follows. The warning from the partner above is similar to mine. I don't know whether there is a problem with our operation or our matlab. Could you please send me the specific process of data import? I still want to give it a try.
--------------------------------------------------------
16-Dec-2024 11:42:48 - Running 'Import'
Warning: The escape character '\E' is invalid. For supported special characters, see 'doc sprintf'.
> Location: pspm_import (Line 232)
Location: pspm_cfg_run_import (line 117)
Location: cfg_run_cm (Line 29)
Location: cfg_util>local_runcj (line 1708)
Location: cfg_util (line 968)
Location: cfg_ui>MenuFileRun_Callback (Line 705)
Location: gui_mainfcn (line 95)
Location: cfg_ui (line 53)

» Importing E:16-Dec-2024 11:42:49 - Failed  'Import'
Unrecognized field name "labels".
In the file "E: \ toolkit \ PsPM \ PsPM_v6 \ pspm_get_acqmat 1.2 m" (??? , function "pspm_get_acqmat" at line 44.
In the file "E: \ toolkit \ PsPM \ PsPM_v6 \ pspm_import 1.2 m" (??? , function "pspm_import" at line 246.
In the file "E: \ toolkit \ PsPM \ PsPM_v6 \ pspm_cfg \ pspm_cfg_run_import 1.2 m" (??? , function "pspm_cfg_run_import" at line 117.

The following modules did not run:
Failed: Import
If the problem can be fixed without modifying the job, the computation can be resumed by running
cfg_util('cont',1)
from the MATLAB command line.

Best wishes
Keyu Li

**Reply by dominikbach** [reply]
Thank you. I suspect that the selected data file (`1.mat` in the screenshot you sent) is not the correct file. I note the file you sent me via email was called `sub1.mat`; and I could import it.

**Reply by yooow-free** [reply]
Thank you very much. I think I know what the problem is. I tried again. It seems that physiological data can be imported without mark data, but I can't import SCR data and mark information at the same time. Is it because my mark was incorrectly loaded? I don't know much about the extraction of mark information, but I only wrote a simple code to extract the mark in the digital channel. Could you please send me the correct acq data containing mark data? I'm not sure how to import it at the same time or in what format. Thank you very much for your patient help, best wishes.

**Reply by dominikbach** [reply]
I have no difficulties importing the file you sent. Matlabbatch script appended below, and screenshot of the imported file attached. 
![screenshot](https://github.com/user-attachments/assets/40e8877b-560d-46ee-9bdb-eae55b9b423c)

```
clear matlabbatch
pspm_jobman('initcfg');
matlabbatch{1}.pspm{1}.prep{1}.import.datatype.acqmat.datafile = {'sub1.mat'};
matlabbatch{1}.pspm{1}.prep{1}.import.datatype.acqmat.importtype{1}.scr.chan_nr.chan_nr_spec = 3;
matlabbatch{1}.pspm{1}.prep{1}.import.datatype.acqmat.importtype{1}.scr.scr_transfer.none = true;
matlabbatch{1}.pspm{1}.prep{1}.import.datatype.acqmat.importtype{2}.marker.chan_nr.chan_nr_spec = 6;
matlabbatch{1}.pspm{1}.prep{1}.import.datatype.acqmat.importtype{2}.marker.flank_option = 'default';
matlabbatch{1}.pspm{1}.prep{1}.import.overwrite = 2;
pspm_jobman('run', matlabbatch);
```

**Comment by yooow-free** [comment]
Thanks, I can import SCR data, but I can't import mark. As far as I know, the tag information collected by biopac is stored in multiple channels, so when importing the tag I'm not sure whether to select a specific channel or search.
![图片](https://github.com/user-attachments/assets/f85cda3c-1849-4ede-8609-e1c3e6ff5a98)
![图片](https://github.com/user-attachments/assets/83439598-b730-4a26-9a95-183d4aae1f34)

**Reply by dominikbach** [reply]
If there are multiple channels then you cannot use the "search" option (because PsPM would not know which of the multiple channels to use). So, you need to specify each of the channels individually. If need be, they can later be combined into one, in a separate step.

**Comment by yooow-free** [comment, marked as answer]
Dear Prof. Dominik,
I can now import the data and marks. In the past, I always chose not to overwrite existing files when importing data, so I did not get the desired result. Now I have sorted out the mark of mat data and got the result I want. Thank you very much for your help. The following is my result:
![图片](https://github.com/user-attachments/assets/3869bd93-4f20-4bd2-8381-2ef2c7b447f3)
